### PR TITLE
Personal favs

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/activities/FavoritesTreeFragment.java [hunk 1]
```diff
@@ -747,14 +752,37 @@ public void synchronizeGroups() {
 			for (FavoriteGroup key : gs) {
 				boolean empty = true;
 				if (flt == null || flt.contains(key)) {
-					empty = false;
-					favoriteGroups.put(key, new ArrayList<>(key.points));
+					if (key.name.equals(PersonalFavouritePoint.PERSONAL)) {
+						ArrayList<FavouritePoint> list = new ArrayList<>();
+						for (FavouritePoint p : key.points) {
+							if (p.getName().equals(PersonalFavouritePoint.PointType.PARKING.name())) {
+								if (parkingPluginEnable) {
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/activities/FavoritesTreeFragment.java]: Strange code. I guess plugin should be asked at the FavsDB.

File: OsmAnd/src/net/osmand/plus/base/FavoriteImageDrawable.java [hunk 2]
```diff
@@ -38,11 +44,20 @@
...
 	private ColorFilter colorFilter;
 	private ColorFilter grayFilter;
+	private Drawable[] personalPointBitmaps;
 
 	public FavoriteImageDrawable(Context ctx, int color, boolean withShadow, boolean synced) {
 		this.withShadow = withShadow;
 		this.synced = synced;
 		Resources res = ctx.getResources();
+		personalPointBitmaps = new Drawable[]{
+				UiUtilities.tintDrawable(ResourcesCompat.getDrawable(res, R.drawable.ic_action_home_dark, null),
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/base/FavoriteImageDrawable.java]: Move R.drawable.ic_action_home_dark to PointType

File: OsmAnd/src/net/osmand/plus/base/FavoriteImageDrawable.java [hunk 3]
```diff
@@ -117,7 +142,7 @@ public void drawBitmapInCenter(Canvas canvas, float x, float y, boolean history)
 
 	@Override
 	public int getOpacity() {
-		return 0;
+		return PixelFormat.UNKNOWN;
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/base/FavoriteImageDrawable.java]: Why opacity unknown?

File: OsmAnd/src/net/osmand/plus/base/FavoriteImageDrawable.java [hunk 4]
```diff
@@ -132,23 +157,27 @@ public void setColorFilter(ColorFilter cf) {
 
 	private static TreeMap<Integer, FavoriteImageDrawable> cache = new TreeMap<>();
 
-	private static FavoriteImageDrawable getOrCreate(Context a, int color, boolean withShadow, boolean synced) {
+	public static FavoriteImageDrawable getOrCreate(Context a, int color, boolean withShadow, boolean synced, int position) {
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/base/FavoriteImageDrawable.java]: position?

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/editors/SelectCategoryDialogFragment.java [hunk 5]
```diff
@@ -79,14 +78,18 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
 		if (gpxFile != null) {
 			if (gpxCategories != null) {
 				for (Map.Entry<String, Integer> e : gpxCategories.entrySet()) {
-					String categoryName = e.getKey();
-					addCategory(activity, ll, categoryName, e.getValue());
+					if (!e.getKey().equals(getContext().getString(R.string.personal_category_name))) {
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/editors/SelectCategoryDialogFragment.java]: Why compare with localized name? Is SelectCategoryDialogFragment used for favs only?

File: OsmAnd/src/net/osmand/plus/search/listitems/QuickSearchListItem.java [hunk 6]
```diff
@@ -102,6 +102,8 @@ public static String getName(OsmandApplication app, SearchResult searchResult) {
 			case LOCATION:
 				LatLon latLon = searchResult.location;
 				return PointDescription.getLocationNamePlain(app, latLon.getLatitude(), latLon.getLongitude());
+			case FAVORITE:
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/search/listitems/QuickSearchListItem.java]: Why it needed?

File: OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java [hunk 7]
```diff
@@ -225,6 +346,91 @@ public boolean addFavourite(FavouritePoint p, boolean saveImmediately) {
...
+		final FavouritePoint workPoint = getWorkPoint();
+		if (workPoint != null && (workPointRequest == null ||
+				!workPointRequest.getLatLon().equals(new LatLon(workPoint.getLatitude(), workPoint.getLongitude())))) {
+			cancelWorkPointAddressRequest();
+			workPointRequest = new GeocodingLookupService.AddressLookupRequest(
+					new LatLon(workPoint.getLatitude(), workPoint.getLongitude()), new GeocodingLookupService.OnAddressLookupResult() {
+				@Override
+				public void geocodingDone(String address) {
+					workPoint.setDescription(address);
+					targetPointsHelper.updateRouteAndRefresh(false);
```
[the remark this answers is not in the document]
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java]: HomeWorkCard doesn't updates without it

**Pull request merged**